### ISSUE TITLE
ViewString on double cosets returns "<object>"

### ISSUE THREAD

**Issue by matsen** [gap-system/gap#29]
@hulpke and I have been discussing this on email and he suggested I post it here. 

The intention of the string and print functions is described in the manual, “ViewString returns a string which would be displayed by ViewObj (6.3-5) for an object”.

This is not true for double cosets as of GAP 4.7.5 of 24-May-2014:

```
gap> dc := DoubleCoset(Group( [ (1,2) ] ),(2,4),Group( [ (3,4), (1,2), (1,3)(2,4) ] ));;
gap> ViewObj(dc);
DoubleCoset(Group( [ (1,2) ] ),(2,4),Group( [ (3,4), (1,2), (1,3)(2,4) ] ))
gap> ViewString(dc);
"<object>"
gap>
```

I worked around this by writing to a string stream:
```
MyPrintToString := function(obj)
    local str, out;
    str := "";;
    out := OutputTextString( str, true );;
    PrintTo(out, obj);
    CloseStream(out);
    return str;
end;;
```

**Comment by alex-konovalov** [comment]
Pull request #30 was merged, let's see how it will pass further tests...

**Comment by alex-konovalov** [comment]
The following diff appeared in the test of manual examples:
```
########> Diff in [ "./group.xml", 174, 181 ]
# Input is:
coset := PreImages( hom, last );
# Expected output:
RightCoset(<group of type 2^3>,(2,8,6,7,3,4,5))
# But found:
RightCoset(Group( [ (1,2)(3,4)(5,6)(7,8), (1,3)(2,4)(5,7)(6,8), (1,5)(\
2,6)(3,7)(4,8) ] ),(2,8,6,7,3,4,5))
########
```
@markuspf, @hulpke, what do you think? Shall the old behaviour be preserved? If we are changing to a new one, would be good to use line break hints to avoid `\`s.

**Comment by markuspf** [comment]
On Mon, Mar 16, 2015 at 02:37:43PM -0700, Alexander Konovalov wrote:
> The following diff appeared in the test of manual examples:
> 
> ########> Diff in [ "./group.xml", 174, 181 ]
> # Input is:
> coset := PreImages( hom, last );
> # Expected output:
> RightCoset(<group of type 2^3>,(2,8,6,7,3,4,5))
> # But found:
> RightCoset(Group( [ (1,2)(3,4)(5,6)(7,8), (1,3)(2,4)(5,7)(6,8), (1,5)(\
> 2,6)(3,7)(4,8) ] ),(2,8,6,7,3,4,5))
> ########
> 
> @markuspf, @hulpke, what do you think? Shall the old behaviour be preserved? If
> we are changing to a new one, would be good to use line break hints to avoid \
> s.


The old behaviour is the better one, because it gives more precise
information. I'll have to see how to at these line break hints to the output.

**Comment by markuspf** [comment]
This has already been addressed in 80a2b93c5
